### PR TITLE
Feat/4477 clean up tests part6

### PR DESCRIPTION
**Pull request by mirogaudi**

### CODE REVIEW

File: spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/server/notify/CompositeNotifierTest.java [hunk 1]
```diff
@@ -30,18 +32,19 @@
...
 	private static final InstanceEvent APP_DOWN = new InstanceStatusChangedEvent(InstanceId.of("-"), 0L,
 			StatusInfo.ofDown());
 
-	@Test
-	public void should_throw_for_invariants() {
-		assertThatThrownBy(() -> new CompositeNotifier(null)).isInstanceOf(IllegalArgumentException.class);
+	@ParameterizedTest
+	@NullSource
+	void should_throw_for_invariants(Iterable<Notifier> delegates) {
+		assertThatThrownBy(() -> new CompositeNotifier(delegates)).isInstanceOf(IllegalArgumentException.class);
```
Comment (erikpetzold) [inline spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/server/notify/CompositeNotifierTest.java]: why the parameterized with null? for me it makes the test more complicated

File: spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/server/notify/MailNotifierTest.java [hunk 2]
```diff
@@ -170,15 +169,15 @@ public void should_not_send_on_wildcard_ignore() {
 	}
 
 	@Test
-	public void should_not_propagate_error() {
-		Notifier notifier = new AbstractStatusChangeNotifier(repository) {
+	void should_not_propagate_error() {
+		Notifier abstractStatusChangeNotifier = new AbstractStatusChangeNotifier(repository) {
```
Comment (erikpetzold) [inline spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/server/notify/MailNotifierTest.java]: Why use the type name as variable name here?

File: spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/server/notify/filter/FilteringNotifierTest.java [hunk 3]
```diff
@@ -31,36 +33,38 @@
...
 		when(repository.find(instance.getId())).thenReturn(Mono.just(instance));
 	}
 
-	@Test
-	public void test_ctor_assert() {
-		Assertions.assertThatThrownBy(() -> new FilteringNotifier(null, repository))
+	@ParameterizedTest
+	@NullSource
+	void test_ctor_assert(Notifier delegate) {
+		Assertions.assertThatThrownBy(() -> new FilteringNotifier(delegate, repository))
```
Comment (erikpetzold) [inline spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/server/notify/filter/FilteringNotifierTest.java]: same here

File: spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/server/notify/MailNotifierTest.java (hunk 2 above)
Comment (mirogaudi) [reply spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/server/notify/MailNotifierTest.java]: "Rename "notifier" which hides the field declared at line 68." -> Renamed to "statusChangeNotifier"

File: spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/server/notify/CompositeNotifierTest.java (hunk 1 above)
Comment (mirogaudi) [reply spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/server/notify/CompositeNotifierTest.java]: It is the way JUnit 5 suggests to test edge cases: https://docs.junit.org/current/user-guide/#writing-tests-parameterized-tests-sources-null-and-empty

File: spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/server/notify/filter/FilteringNotifierTest.java (hunk 3 above)
Comment (mirogaudi) [reply spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/server/notify/filter/FilteringNotifierTest.java]: It is the way JUnit 5 suggests to test edge cases: https://docs.junit.org/current/user-guide/#writing-tests-parameterized-tests-sources-null-and-empty

File: spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/server/notify/CompositeNotifierTest.java (hunk 1 above)
Comment (hzpz) [reply spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/server/notify/CompositeNotifierTest.java]: That makes sense if the test is _already_ parameterized. In the case above, however, I'm with Erik.

Comment (mirogaudi) [reply spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/server/notify/CompositeNotifierTest.java]: Reverted

File: spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/server/notify/filter/FilteringNotifierTest.java (hunk 3 above)
Comment (mirogaudi) [reply spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/server/notify/filter/FilteringNotifierTest.java]: Reverted
